nucypher/blockchain/eth/actors.py

```diff
@@ -396,241 +392,94 @@ def deploy_network_contracts(self,
-
-            if emitter:
-                paint_deployed_allocations(emitter, allocated, failed)
-
-            csv_filename = f'allocation-summary-{self.deployer_address[:6]}-{maya.now().epoch}.csv'
-            csv_filepath = os.path.join(output_dir, csv_filename)
-            write_deployed_allocations_to_csv(csv_filepath, allocated, failed)
-            if emitter:
-                emitter.echo(f"Saved allocation summary CSV to {csv_filepath}", color='blue', bold=True)
-
-            if failed:
-                self.log.critical(f"FAILED TOKEN ALLOCATION - {len(failed)} allocations failed.")
-
-        return allocation_receipts
-
-    @staticmethod
-    def __read_allocation_data(filepath: str) -> list:
-        with open(filepath, 'r') as allocation_file:
-            if filepath.endswith(".csv"):
-                reader = csv.DictReader(allocation_file)
-                allocation_data = list(reader)
-            else:  # Assume it's JSON by default
-                allocation_data = json.load(allocation_file)
+                bar.update(number_of_deposits)
 
-        # Pre-process allocation data
-        for entry in allocation_data:
-            entry['beneficiary_address'] = to_checksum_address(entry['beneficiary_address'])
-            entry['amount'] = int(entry['amount'])
-            entry['duration_seconds'] = int(entry['duration_seconds'])
-
-        return allocation_data
-
-    def deploy_beneficiaries_from_file(self,
-                                       allocation_data_filepath: str,
-                                       allocation_outfile: str = None,
-                                       emitter=None,
-                                       interactive=None) -> dict:
-
-        allocations = self.__read_allocation_data(filepath=allocation_data_filepath)
-        receipts = self.deploy_beneficiary_contracts(allocations=allocations,
-                                                     allocation_outfile=allocation_outfile,
-                                                     emitter=emitter,
-                                                     interactive=interactive,
-                                                     crash_on_failure=False)
-        # Save transaction metadata
-        receipts_filepath = self.save_deployment_receipts(receipts=receipts, filename_prefix='allocation')
+                if emitter:
+                    emitter.echo()
+                    paint_receipt_summary(emitter=emitter,
+                                          receipt=receipt,
+                                          chain_name=chain_name,
+                                          transaction_type=f'batch_deposit_{number_of_deposits}_stakers')
+
+                batch_deposit_receipts.update({staker: {'batch_deposit': receipt} for staker in deposited_stakers})
+
+                if interactive:
+                    click.pause(info=f"\nPress any key to continue with next batch of allocations")
```

Comment:
If the `deposit_next_batch()` fails, and `crash_on_failure` is `False`, is the same batch potentially retried repeatedly?

If so, that's fine but we should alter the wording of our message for that case.

Reply:
`crash_on_failure` is a remnant from the previous approach. I was undecided about leaving this or not, and I didn't think about this edge case, so I'm getting rid of it. Now if it fails, it gracefully crashes and doesn't continue with the remaining allocations.

Reply:
👍